internal/controller/nonadminbackup_controller.go

```diff
@@ -56,6 +60,10 @@ const (
 	veleroReferenceUpdateRequeue = "NonAdminBackup - Requeue after Status Update with UUID reference"
 	statusUpdateExit             = "NonAdminBackup - Exit after Status Update"
 	statusUpdateError            = "Failed to update NonAdminBackup Status"
+	findSingleVBError            = "Failed to find single VeleroBackup object"
```

Comment:
you need to change the err string here. This might imply that you could not find a single VB for the particular UUID. I would reframe it to some thing like: `Error encountered while retrieving VeleroBackup for NAB during the Delete operation` your get function already has the particular error.

Reply:
Done please check newest commit.